pkg/controller/ingress-shim/sync.go

```diff
@@ -88,24 +114,37 @@ func (c *Controller) Sync(ctx context.Context, ing *extv1beta1.Ingress) error {
 	return nil
 }
 
-func (c *Controller) buildCertificates(ing *extv1beta1.Ingress) (new, update []*v1alpha1.Certificate, _ error) {
-	issuerName, issuerKind := c.issuerForIngress(ing)
-	issuer, err := c.getGenericIssuer(ing.Namespace, issuerName, issuerKind)
-	if err != nil {
-		return nil, nil, err
+func (c *Controller) validateIngress(ing *extv1beta1.Ingress) []error {
+	var errs []error
+	if ing.Annotations != nil {
+		challengeType := ing.Annotations[acmeIssuerChallengeTypeAnnotation]
+		switch challengeType {
+		case "", "http01":
+		case "dns01":
+			providerName := ing.Annotations[acmeIssuerDNS01ProviderNameAnnotation]
+			if providerName == "" {
```

Comment:
@munnerz We use to be able to set `certmanager.k8s.io/acme-challenge-type: "dns01"` without having to set `certmanager.k8s.io/acme-dns01-provider`, in which case, the default dns01 provider would be used - which is still a valid case in the code: https://github.com/jetstack/cert-manager/blob/52ce959356e5c8b9cfb30841b780c33d3eef5cc5/pkg/controller/ingress-shim/sync.go#L312. However, your new validation system will block the ingress from being processed.

Is this expected?